html5lib/_tokenizer.py

```diff
@@ -448,7 +449,7 @@ def tagNameState(self):
                                     "data": "invalid-codepoint"})
             self.currentToken["name"] += "\uFFFD"
         else:
-            self.currentToken["name"] += data
+            self.currentToken["name"] += data.translate(asciiUpper2Lower)
```

Comment:
I'm very skeptical about this being a perf win, versus it being in `emitCurrentToken`. What do the benchmarks say?

Yes, `emitCurrentToken`'s lowercasing becomes redundant in the RCDATA/RAWTEXT/script cases, but I expect the cost of this will negate any gains.

Reply:
That's fair, yep - especially for short element names it seems likely that the `translate` method call overhead (especially if called repeatedly) could negate any benefits provided by simpler comparisons.

I hadn't assessed the performance of this code path separately; it felt worth maintaining consistency but I don't believe there's a noticeable performance change.